core/domain/topic_services_test.py

```diff
@@ -1824,16 +1824,59 @@ def test_update_subtopic_thumbnail_filename(self) -> None:
             self.topic.subtopics[0].thumbnail_size_in_bytes, len(raw_image))
 
     def test_get_topic_id_to_diagnostic_test_skill_ids(self):
+        fractions_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            fractions_id, self.user_id, name='Fractions',
+            url_fragment='fractions', description='Description of fraction',
+            canonical_story_ids=[self.story_id_1, self.story_id_2],
+            additional_story_ids=[self.story_id_3],
+            uncategorized_skill_ids=[self.skill_id_1, self.skill_id_2],
+            subtopics=[], next_subtopic_id=1)
+        changelist = [
+            topic_domain.TopicChange({
+                'cmd': topic_domain.CMD_UPDATE_TOPIC_PROPERTY,
+                'property_name': (
+                    topic_domain.TOPIC_PROPERTY_SKILL_IDS_FOR_DIAGNOSTIC_TEST),
+                'old_value': [],
+                'new_value': [self.skill_id_1]
+            })]
+        topic_services.update_topic_and_subtopic_pages(
+            self.user_id_admin, fractions_id, changelist,
+            'Adds diagnostic test.')
+
+        additions_id = topic_fetchers.get_new_topic_id()
+        self.save_new_topic(
+            additions_id, self.user_id, name='Additions',
+            url_fragment='additions', description='Description of addition.',
+            canonical_story_ids=[self.story_id_1, self.story_id_2],
+            additional_story_ids=[self.story_id_3],
+            uncategorized_skill_ids=[self.skill_id_1, self.skill_id_2],
+            subtopics=[], next_subtopic_id=1)
+        changelist = [
+            topic_domain.TopicChange({
+                'cmd': topic_domain.CMD_UPDATE_TOPIC_PROPERTY,
+                'property_name': (
+                    topic_domain.TOPIC_PROPERTY_SKILL_IDS_FOR_DIAGNOSTIC_TEST),
+                'old_value': [],
+                'new_value': [self.skill_id_2]
+            })]
+        topic_services.update_topic_and_subtopic_pages(
+            self.user_id_admin, additions_id, changelist,
+            'Adds diagnostic test.')
+
         expected_dict = {
-            self.TOPIC_ID: self.topic.skill_ids_for_diagnostic_test
+            fractions_id: [self.skill_id_1],
+            additions_id: [self.skill_id_2]
         }
         self.assertEqual(
             topic_services.get_topic_id_to_diagnostic_test_skill_ids(
-                [self.TOPIC_ID]), expected_dict)
+                [fractions_id, additions_id, 'incorrect_topic_id', None]),
+            expected_dict
+        )
 
         self.assertEqual(
             topic_services.get_topic_id_to_diagnostic_test_skill_ids(
-                'incorrect_topic_id'), {})
+                ['incorrect_topic_id']), {})
```

Comment:
Added code to raise `exception` for incorrect topic ID (the topic ID for which the topic model does not exist)
For the None case i.e., among the list of topic IDs, some elements are None -- do we have to check for that case too? because I think in general they can't happen in reality.
Please provide your suggestion.

Reply:
Yeah, I think you are correct about None. Let's leave that -- thanks!